numba/interpreter.py

```diff
@@ -242,119 +243,169 @@ def _compute_generator_info(self):
-            #  and need not be kept alive accross iterations)
-            live_points.update(loop.exits)
-
-        for block in use_map:
-            live_points.add(block)
-
-        # Now compute a set of *exclusive* blocks covering all paths
-        # stemming from the live points. This ensures that each variable
-        # is del'ed at most once per possible code path.
-        tails = set()
-        # We use the todo list as a stack, so we really iterate in
-        # reverse topo order.
-        todo = list(cfg.topo_sort(live_points))
-        seen = set()
-        infinite_loops = set()
-        while todo:
-            b = todo.pop()
-            if b in seen:
-                continue
-            seen.add(b)
-            block_loops = cfg.in_loops(b)
-            if block_loops:
-                loop = block_loops[0]
-                if loop not in enclosing_loops:
-                    if not loop.exits:
-                        # Fix-up for infinite loops: we temporarily add
-                        # the loop header to the tails, so that the variable
-                        # isn't del'ed before the loop.
-                        tails.add(loop.header)
-                        infinite_loops.add(loop.header)
-                    continue
-            if cfg.descendents(b) & tails:
-                for succ, _ in cfg.successors(b):
-                    if succ not in tails:
-                        todo.append(succ)
+        for offset, ir_block in self.blocks.items():
+            # live vars WITHIN the block will include all the locally
+            # defined variables
+            cur_live_set = live_map[offset] | var_def_map[offset]
+            # vars alive alive in the outgoing blocks
+            outgoing_live_map = dict((out_blk, live_map[out_blk])
+                                     for out_blk, _data
+                                     in self.cfa.graph.successors(offset))
+            # vars to keep alive for the terminator
+            terminator_liveset = set(v.name
+                                     for v in ir_block.terminator.list_vars())
+            # vars to keep alive in the successors
+            combined_liveset = reduce(operator.or_, outgoing_live_map.values(),
+                                      set())
+            # include variables used in terminator
+            combined_liveset |= terminator_liveset
+            # vars that are dead within the block beacuse they are not
+            # propagated to any outgoing blocks
+            intern_set = cur_live_set - combined_liveset
+            intern_dead_map[offset] = intern_set
+            # vars that escape this block
+            escaping_live_set = cur_live_set - intern_set
+            for out_blk, new_live_set in outgoing_live_map.items():
+                # successor should delete the unused escaped vars
+                new_live_set = new_live_set | var_def_map[out_blk]
```

Comment:
Yes, it is safe to delete.  But it breaks the assumption of the live variable tracking of objmode, and the variable could be deleted earlier than it would in cpython.